campaignion_webform_widget/js/widget.js

```diff
@@ -49,18 +49,20 @@ function gaLinkerHandler() {
   }
 }
 
-function registerListeners() {
-  if (window.addEventListener) {
-    window.addEventListener("DOMSubtreeModified", function(){messageParent();}, true);
-    document.addEventListener("DOMContentLoaded", gaLinkerHandler);
-  } else if (window.attachEvent) {
-    window.attachEvent("onDOMSubtreeModified", function(){messageParent();});
-  }
+if (window.addEventListener) {
+  window.addEventListener("DOMContentLoaded", function() {
+    gaLinkerHandler();
+    messageParent();
+    window.addEventListener("DOMSubtreeModified", messageParent, true);
```

Comment:
Why register the "DOMSubtreeModified" handler within the "DOMContentLoaded" handler?

Reply:
I placed it within the `DOMContentLoaded` handler for two reasons:
- The old `registerListeners` function that previously contained it was originally activated on `window.load`; having chosen instead to use the `DOMContentLoaded` event to represent the page load, it seemed to be the most equivalent position for it in the new structure.
- I am not 100% sure if this is the case, but it occurred to me that setting up a `DOMSubtreeModified` listener before the DOM had finished loading might cause a deluge of pointless events... so I kept it within the `DOMContentLoaded` handler in an effort to prevent this (see also comment below).